dex/src/main/java/io/crate/data/ForwardingBatchIterator.java

```diff
@@ -29,10 +29,15 @@
  */
 public abstract class ForwardingBatchIterator implements BatchIterator {
 
+    protected abstract BatchIterator delegate();
```

Comment:
was it intentional to move the abstract method above of the constructor? I don't mind, though we usually keep methos below constructors.